packages/block-library/src/navigation/edit/index.js

```diff
@@ -152,13 +168,29 @@ function Navigation( {
 				( block ) => block.name === 'core/navigation-submenu'
 			);
 
+			const { canUser, hasFinishedResolution } = select( coreStore );
+
 			return {
 				hasSubmenus: firstSubmenu,
 				innerBlocks: blocks,
 				isInnerBlockSelected: hasSelectedInnerBlock( clientId, true ),
+				canUserUpdateNavigationEntity: ref
+					? canUser( 'update', 'navigation', ref )
```

Comment:
It was mentioned elsewhere, but 'update' is a `PUT` request, which is never actually something we do. I think this should be 'create' which represents a `POST`.

Reply:
So the permissions handling is pretty confusing. Here's my understanding.

- `canUser('update')` checks for a `PUT` included in the `Allow` header in the response from the `/wp/navigation/{ID}` endpoint.
- [the `Allow` header is set](https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api.php#L797) based on the `permission_callback` of the endpoint.
- that endpoint defers to the [`WP_REST_Posts_Controller`](https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php).
- if the endpoint is called with an `ID` then it will be one [the `EDITABLE` methods](https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api/class-wp-rest-server.php#L39) that are [the handlers](https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L122-L127). 
- the `permission_callback` is [`update_item_permissions_check`](https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L766) (which is what is called to determine the `Allow` header.

Therefore I _think_ using `update` here is valid. Indeed this is [the pattern also in use by the Site Logo block](https://github.com/WordPress/gutenberg/blob/4cedaf0c4abb0aeac4bfd4289d63e9889efe9733/packages/block-library/src/site-logo/edit.js#L329).

Reply:
What @getdave has written here is correct and this looks good to me. Basically anyone can create a post in WordPress. But being able to publish or edit post is what is needed here. Even a contributor can create a post, this just means they can create a draft / pending post. Knowing that isn't especially useful here.